### PR TITLE
feat: first pass for application mgmt apis

### PR DESCRIPTION
**Pull request by tomkit**
- add missing service and endpoints for customer, integration, connection concepts
- add middleware for capturing params

TODO
- revise customer schema
- openapi schema
- mgmt ui

### CODE REVIEW

File: apps/api/routes/customer/index.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { customerMiddleware } from '@/middleware/customer';
+import { Request, Response, Router } from 'express';
+import integration from './integration';
+
+const { customerService } = getDependencyContainer();
+
+export default function init(app: Router): void {
+  const customerRouter = Router();
+
+  app.get('/customers', async (req: Request, res: Response) => {
```
Comment (lucasmarshall) [inline apps/api/routes/customer/index.ts:11]: Let's type these?

File: apps/api/routes/customer/integration/connection/index.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { getDependencyContainer } from '@/dependency_container';
+import { connectionMiddleware } from '@/middleware/connection';
+import { Request, Response, Router } from 'express';
+
+const { connectionService } = getDependencyContainer();
+
+export default function init(app: Router): void {
+  app.get('/connections', async (req: Request, res: Response) => {
```
Comment (lucasmarshall) [inline apps/api/routes/customer/integration/connection/index.ts:8]: Same as above, need types

File: apps/api/routes/customer/integration/index.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { getDependencyContainer } from '@/dependency_container';
+import { integrationMiddleware } from '@/middleware/integration';
+import { Request, Response, Router } from 'express';
+import connection from './connection';
+
+const { integrationService } = getDependencyContainer();
+
+export default function init(app: Router): void {
+  app.get('/integrations', async (req: Request, res: Response) => {
```
Comment (lucasmarshall) [inline apps/api/routes/customer/integration/index.ts:9]: Same as above, need types

File: packages/core/types/customer.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  id: string;
+  createdAt: Date;
+  updatedAt: Date;
+
+  // TODO: add fields
+};
+export type BaseCustomer = Customer & {
+  // TODO: add fields
+};
+export type BaseCustomerCreateParams = Omit<Omit<Omit<BaseCustomer, 'id'>, 'createdAt'>, 'updatedAt'>;
```
Comment (khennes) [inline packages/core/types/customer.ts:11]:
```suggestion
export type BaseCustomerCreateParams = Omit<BaseCustomer, 'id' | 'createdAt' | 'updatedAt'>;
```

File: apps/api/routes/customer/index.ts (hunk 1 above)
Comment (tomkit) [reply apps/api/routes/customer/index.ts:11]: These come from open api -> ts, correct?

I was going to add OpenApi specs in another PR and then I'll add the generic types here

**Review by lucasmarshall — DISMISSED** [review]

**Review by lucasmarshall — APPROVED** [review]

**Pull request merged**